business/istio_validations.go

```diff
@@ -92,11 +97,14 @@ func (in *IstioValidationsService) GetValidations(ctx context.Context, namespace
 	} else {
 		go in.fetchAllWorkloads(ctx, &workloadsPerNamespace, &namespaces, errChan, &wg)
 	}
-	go in.fetchNonLocalmTLSConfigs(&mtlsDetails, errChan, &wg)
-	if service != "" {
-		go in.fetchServices(ctx, &services, namespace, errChan, &wg)
+
+	if istioApiEnabled {
+		go in.fetchNonLocalmTLSConfigs(&mtlsDetails, errChan, &wg)
+		if service != "" {
+			go in.fetchServices(ctx, &services, namespace, errChan, &wg)
+		}
+		go in.fetchRegistryServices(&registryServices, errChan, &wg)
```

Comment:
We don't want to fetch services or non local mtls configs when the registry is disabled? Seems like we only want to exclude the registry services?

Reply:
Updated